pitch.md

```diff
@@ -14,31 +23,170 @@ _What need is met by your idea? Who are you helping? What is that person's
+distinct ideas may be spatially close. If input is spatial then this can result
+in geometric imprecision. For example, consider drawing a line from a triangle
+vertex to the midpoint of the opposing side vs bisecting that angle. Often
+these two actions are spatially similar (the lines end up close to each other).
+
+Current solutions to the problem fail to be both easy to use and geometrically
+precise. `tkz-euclid` is geometrically precise but very cryptic, and there are
+various visual construction engines which are easy to use but rely on spatial
+input and are thus geometrically imprecise in some cases.
+
+Furthermore, none of the above systems allow the user to see what steps they
+might take next.
+
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+Some parts of this solution are clearly a language. The way the user interacts
+with the system clearly must be a language. If they use some sort of textual
+system like
+
+```
+Line from A to B
+Line from A to C
+Bisect Angle C A B
+```
+
+then there is a language writen there with all of the nouns and verbs we
+expect.
+
+But there is also more. Under the hood, the system must have some sort of idea
+of how to represent these geometric ideas. A common form from which the
+description system, the rendering system, and the extrapolation system can all
+be build.
+
+I think that this language - this core representation of geometric ideas is the
+real DSL behind this project.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I've always been somewhat of a geometry enthusiast, which is probably the
+real origin of my interest in this project. I enjoy thinking about geometric
+ideas and think that I would have a lot of fun doing that for the next few
+weeks.
+
+That's not all though. I also generally have a lot of trouble using computers
+to foster creativity. I'm a very paper and pencil sort of guy. I think
+designing a system which has to potential to make computers a better platform
+for exploring ideas (even in the limited domain of classical constructions)
+would be worth putting time into.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain? Be careful
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+#### User Language
+
+The User Language is the way in which users write constructions.
+
+Taken from my project ideas writeup:
+
+> [Users] would write statements like:
+>
+> ```
+> circle x with center C
+> line y tangent to x at point D
+> line z through points C and D
+> ```
+>
+> I really like language idea because it mirrors the way that mathematicians
+> write about geometry. Who hasn't seen something like:
+>
+> Let x be a circle centered at C. Let y be a line tangent to x at point D. Let z
+> be a line through C and D. ...more proof...
```

Comment:
Indeed. As I work on this more I suspect that some some of prover would make this a much better product. Perhaps it would also be too much for me to do though.